tools/crm_resource_runtime.c

```diff
@@ -1638,18 +1638,14 @@ wait_till_stable(int timeout_ms, cib_t * cib)
 }
 
 int
-cli_resource_execute(resource_t *rsc, const char *requested_name,
-                     const char *rsc_action, GHashTable *override_hash,
-                     int timeout_ms, cib_t * cib, pe_working_set_t *data_set)
+cli_resource_execute_from_params(const char *rsc_name, const char *rsc_class,
+                                 const char *rsc_prov, const char *rsc_type,
+                                 const char *rsc_action, GHashTable *params,
+                                 GHashTable *override_hash, int timeout_ms)
 {
     int rc = pcmk_ok;
-    svc_action_t *op = NULL;
-    const char *rid = NULL;
-    const char *rtype = NULL;
-    const char *rprov = NULL;
-    const char *rclass = NULL;
     const char *action = NULL;
-    GHashTable *params = NULL;
+    svc_action_t *op = NULL;
 
     if (safe_str_eq(rsc_action, "validate")) {
         action = "validate-all";
```

Comment:
My thinking here was to preserve what was happening in cli_resource_excecute, but just move it into the new function.  I also wanted the two functions to act the same - if you know how to use cli_resource_execute, you'll know how to use the new one.  If that doesn't matter, I can just move it all back to cli_resource_execute and the new function will just expect you to pass what you mean.

Reply:
It's all internal to crm_resource, so there's no problem moving things around as needed. A comment with the requirement is sufficient.

cli_resource_execute() needs the remapping because it uses both action and rsc_action in log messages, and uses action in pe_get_configured_timeout() and resources_action_create().